server/api/users/users.controller.ts

```diff
@@ -38,3 +38,43 @@ export const getUserById: RequestHandler<{id: string}, UserProfile> = async (req
 
   res.json(user);
 };
+
+interface UpdatableFields {
+  bio?: string;
+  twitter_username?: string;
+  linkedin_url?: string;
+  github_username?: string;
+  website?: string;
+}
+
+export const updateUserById: RequestHandler<{ id: string }, string, UpdatableFields> = async (req, res) => {
+
+  const fieldsToUpdate = [
+    'bio',
+    'twitter_username',
+    'linkedin_url',
+    'github_username',
+    'website',
+  ];
+
+  for (const field of fieldsToUpdate) {
+    if (field in req.body) {
+      break;
+    }
+    throw new BadRequestError('Request body should contain at least one updatable field.');
```

Comment:
I believe this can be simplified using a conditional and `Object.keys(toUpdate).length` or even `_.size(toUpdate)` after the `pick`.